js/mcqView.js

```diff
@@ -1,175 +1,86 @@
-      });
-      this.listenTo(this.model.getChildren(), {
-        "change:_isActive": this.update
-      });
-    },
-
-    resetQuestionOnRevisit: function() {
-      this.resetQuestion();
-    },
-
-    setupQuestion: function() {
-      this.model.setupRandomisation();
-    },
-
-    onQuestionRendered: function() {
-      this.setReadyStatus();
-      this.update();
-    },
-
-    onKeyPress: function(event) {
-      if (event.which !== 13) return;
-      //<ENTER> keypress
-      this.onItemSelect(event);
-    },
-
-    onItemFocus: function(event) {
-      if (!this.model.isInteractive()) return;
-
-      this.$('.js-item-label[for='+$(event.currentTarget).attr('id')+']').addClass('is-highlighted');
-    },
-
-    onItemBlur: function(event) {
-      this.$('.js-item-label[for='+$(event.currentTarget).attr('id')+']').removeClass('is-highlighted');
-    },
-
-    onItemSelect: function(event) {
-      if (!this.model.isInteractive()) return;
-
-      var index = $(event.currentTarget).data('adapt-index');
-      var itemModel = this.model.getItem(index);
-      var shouldSelect = !itemModel.get("_isActive");
-
-      if (this.model.isSingleSelect()) {
-        // Assume a click is always a selection
-        shouldSelect = true;
-        this.model.resetActiveItems();
-      } else if (shouldSelect && this.model.isAtActiveLimit()) {
-        // At the selection limit, deselect the last item
-        this.model.getLastActiveItem().toggleActive(false);
-      }
-
-      // Select or deselect accordingly
-      itemModel.toggleActive(shouldSelect);
-    },
-
-    // Blank method to add functionality for when the user cannot submit
-    // Could be used for a popup or explanation dialog/hint
-    onCannotSubmit: function() {},
-
-    // This is important and should give the user feedback on how they answered the question
-    // Normally done through ticks and crosses by adding classes
-    showMarking: function() {
-      this.update();
-    },
-
-    // Used by the question view to reset the look and feel of the component.
-    resetQuestion: function() {
+import QuestionView from 'core/js/views/questionView';
+
+class McqView extends QuestionView {
+
+  initialize(...args) {
+    super.initialize(...args);
+    this.listenTo(this.model, {
+      "_isEnabled _isComplete _isSubmitted _isCorrectAnswerShown": this.changed
+    });
+    this.listenTo(this.model.getChildren(), {
+      'all': this.changed
+    });
+    this.model.isCorrectAnswerShown = false;
+  }
+
+  resetQuestionOnRevisit() {
+    this.resetQuestion();
+  }
+
+  setupQuestion() {
+    this.model.setupRandomisation();
+  }
+
+  onQuestionRendered() {
+    this.setReadyStatus();
+  }
+
+  onKeyPress(event) {
+    if (event.which !== 13) return;
+    //<ENTER> keypress
+    this.onItemSelect(event);
+  }
+
+  onItemFocus(event) {
+    if (!this.model.isInteractive()) return;
+
+    this.$('.js-item-label[for='+$(event.currentTarget).attr('id')+']').addClass('is-highlighted');
```

Comment:
template string might be a bit nicer here?
```suggestion
    this.$(`.js-item-label[for=${$(event.currentTarget).attr('id'}]`).addClass('is-highlighted');
```

Reply:
alternatively:
```suggestion
    const id = $(event.currentTarget).attr('id');
    this.$(`.js-item-label[for=${id}]`).addClass('is-highlighted');
```